src/app/components/mirror/mirror.component.ts

```diff
@@ -16,5 +16,5 @@ export class MirrorComponent implements OnInit {
     this.page = new PagesService(this.router);
```

Comment:
you can register service in @Component decorator and a new service instance will be defined for each component